src/Datadog.Trace.ClrProfiler.Managed/Emit/DynamicMethodBuilder.cs

```diff
@@ -329,18 +336,22 @@ internal static class DynamicMethodBuilder<TDelegate>
                 }
             }
 
-            if (methodInfo.IsStatic)
+            if (callOpCode == OpCodeValue.Call || methodInfo.IsStatic)
```

Comment:
This is where we use the new opcode value when emitting IL. Most integrations will still pass the default opcode (`CALLVIRT`), so we still check if it's a static method and use `CALL`.